### PR TITLE
tests: shellchecks part 8

### PR DESCRIPTION
**Pull request by bboozzoo**
Part 8

### CODE REVIEW

File: tests/main/snap-auto-import-asserts-spools/task.yaml [hunk 1]
```diff
@@ -36,17 +37,17 @@ execute: |
...
     umount /mnt
     systemctl start snapd.service snapd.socket
 
-    echo "`snap auto-import` reads from the auto-import dir"
+    echo "'snap auto-import' reads from the auto-import dir"
     snap auto-import
     snap known account-key | MATCH "name: test-store"
 
-    nr=$(ls /run/snapd/auto-import|wc -l)
+    nr=$(find /run/snapd/auto-import -maxdepth 1 -mindepth 1 |wc -l)
```
Comment (mvo5) [inline tests/main/snap-auto-import-asserts-spools/task.yaml]: Curious, why do we need mindepth 1 here?

File: tests/main/prepare-image-grub/task.yaml [hunk 2]
```diff
@@ -47,6 +47,7 @@ execute: |
     export SNAPPY_FORCE_SAS_URL=http://$STORE_ADDR
 
     echo Running prepare-image
+    #shellcheck disable=SC2086
```
Comment (mvo5) [inline tests/main/prepare-image-grub/task.yaml]: Is this needed? I looked it up in the wiki and it seems to be doing something that is not present in the following line. Also when I run my (18.04) shellcheck against this file I get no complains.

File: tests/main/snap-connect/task.yaml [hunk 3]
```diff
@@ -27,7 +28,7 @@ execute: |
     snap connect home-consumer:home system:home
 
     OUTPUT=$(snap change --last=connect)
-    if [ ! $(echo "$OUTPUT" | wc -l) -eq 1 ]; then
+    if [ ! "$(echo "$OUTPUT" | wc -l)" -eq 1 ]; then
```
Comment (mvo5) [inline tests/main/snap-connect/task.yaml]: drive-by: could we use `if [ "$()" -ne 1 ]; then` here instead of the `! -eq` .

File: tests/main/snap-info/task.yaml [hunk 4]
```diff
@@ -15,16 +15,17 @@ restore: |
 
 execute: |
     echo "With no arguments, errors out"
-    snap info && exit 1 || true
+    ! snap info
 
     echo "With one non-snap argument, errors out"
-    snap info /etc/passwd && exit 1 || true
+    ! snap info /etc/passwd
```
Comment (mvo5) [inline tests/main/snap-info/task.yaml]: nice

File: tests/main/snap-auto-import-asserts-spools/task.yaml (hunk 1 above)
Comment (bboozzoo) [reply tests/main/snap-auto-import-asserts-spools/task.yaml]: That's because:
```
$ mkdir foo
$ ls foo
$ find foo -maxdepth 1 -mindepth 1
$ find foo -maxdepth 1
foo
```
Either that or the `if` below would need `"$nr" -gt 1`

File: tests/main/snap-connect/task.yaml [hunk 5]
```diff
@@ -27,7 +28,7 @@ execute: |
     snap connect home-consumer:home system:home
 
     OUTPUT=$(snap change --last=connect)
-    if [ ! $(echo "$OUTPUT" | wc -l) -eq 1 ]; then
+    if [ "$(echo "$OUTPUT" | wc -l)" -ne 1 ]; then
```
Comment (stolowski) [inline tests/main/snap-connect/task.yaml]: Nice, reads better that way.
